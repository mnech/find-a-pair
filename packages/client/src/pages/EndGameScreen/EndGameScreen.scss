$font-family: 'Caveat', cursive;

.endGameTitle {
  font-family: $font-family;
  font-size: 200px;
}

.endGameScore {
  font-family: $font-family;
  font-size: 60px;
}
