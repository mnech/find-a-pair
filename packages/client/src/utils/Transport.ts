import axios from 'axios';
import { Methods } from '../api/consts';

export type TRequest = {
  method?: Methods
  url: string
  data?: unknown
};

export default class Transport {
  protected axios = axios;

  constructor() {
    this.axios.defaults.withCredentials = true;
  }

  public request({ method = Methods.Get, url, data }: TRequest) {
    return this.axios({ method, url, data });
  }
}
