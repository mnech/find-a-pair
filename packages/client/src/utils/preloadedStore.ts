import {
  limitPage,
  paginationMin,
  ratingFieldName,
} from '../models/Leaderboard';

export const preloadedState = {
  leaderboard: {
    //пока заглушка
    data: [
      {
        name: 'User',
        score: 1,
      },
      {
        name: 'Jane',
        score: 2,
      },
      {
        name: 'James',
        score: 7,
      },
      {
        name: 'Luis',
        score: 10,
      },
    ],
    request: {
      ratingFieldName: ratingFieldName,
      cursor: paginationMin,
      limit: limitPage,
    },
  },
};
