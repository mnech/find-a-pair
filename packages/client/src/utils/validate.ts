import { Regexp } from '../consts';

export const regexpTest = (type: keyof typeof Regexp, str: string) => {
  return Regexp[type].test(str);
};
