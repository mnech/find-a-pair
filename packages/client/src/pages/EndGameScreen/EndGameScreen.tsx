import React from 'react';
import { Link } from 'react-router-dom';
import { routes } from '../../models/App';
import './EndGameScreen.scss';
import { useSelector } from 'react-redux';
import { RootState } from '../../store';

export const EndGameScreen = () => {
  const score = useSelector((state: RootState) => state.game.score);
  return (
    <div className="container-md mt-5 w-50 h-auto d-flex flex-column align-items-center">
      <p className="endGameTitle">Конец игры</p>
      <p className="endGameScore">Счёт: {score}</p>
      <Link to={routes.game} className="btn btn-primary">
        Попробовать еще раз?
      </Link>
      <Link to={routes.leaderboard}>Лидерборд</Link>
      <Link to={routes.profile}>Профиль</Link>
    </div>
  );
};
