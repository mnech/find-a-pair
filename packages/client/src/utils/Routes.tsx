import { ErrorBoundary } from './ErrorBoundary';
import { Auth } from '../pages/Auth/Auth';
import { Leaderboard } from '../pages/Leaderboard/Leaderboard';
import Game from '../pages/Game/Game';
import Forum from '../pages/Forum/Forum';
import Profile from '../pages/Profile/Profile';
import { StartGameScreen } from '../pages/StartGameScreen/StartGameScreen';
import Signup from '../pages/SignUp/SignUp';
import { routes } from '../models/App';
import { EndGameScreen } from '../pages/EndGameScreen/EndGameScreen';
export const staticRoutes = [
  {
    path: routes.auth,
    errorElement: <ErrorBoundary />,
    children: [
      {
        index: true,
        path: routes.auth,
        element: <Auth />,
      },
      {
        path: routes.profile,
        element: <Profile />,
      },
      {
        path: routes.signUp,
        element: <Signup />,
      },
      {
        path: routes.leaderboard,
        element: <Leaderboard />,
      },
      {
        path: routes.forum,
        element: <Forum />,
      },
      {
        path: routes.game,
        element: <Game />,
      },
      {
        path: routes.startGame,
        element: <StartGameScreen />,
      },
      {
        path: routes.endGame,
        element: <EndGameScreen />,
      },
    ],
  },
];
