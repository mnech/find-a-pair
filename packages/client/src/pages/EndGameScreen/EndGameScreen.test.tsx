import { renderWithProviders } from '../../utils/configureStore';
import { EndGameScreen } from './EndGameScreen';

test('Auth renders correctly', () => {
  const { asFragment } = renderWithProviders(<EndGameScreen />);
  expect(asFragment()).toMatchSnapshot();
});
