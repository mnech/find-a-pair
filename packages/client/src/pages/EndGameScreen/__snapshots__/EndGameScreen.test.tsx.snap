// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container-md mt-5 w-50 h-auto d-flex flex-column align-items-center"
  >
    <p
      class="endGameTitle"
    >
      Конец игры
    </p>
    <p
      class="endGameScore"
    >
      Счёт: 40
    </p>
    <a
      class="btn btn-primary"
      href="/game"
    >
      Попробовать еще раз?
    </a>
    <a
      href="/leaderboard"
    >
      Лидерборд
    </a>
    <a
      href="/"
    >
      Профиль
    </a>
  </div>
</DocumentFragment>
`;

exports[`End Game renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container-md mt-5 w-50 h-auto d-flex flex-column align-items-center"
  >
    <p
      class="endGameTitle"
    >
      Конец игры
    </p>
    <p
      class="endGameScore"
    >
      Счёт: 40
    </p>
    <a
      class="btn btn-primary"
      href="/game"
    >
      Попробовать еще раз?
    </a>
    <a
      href="/leaderboard"
    >
      Лидерборд
    </a>
    <a
      href="/"
    >
      Профиль
    </a>
  </div>
</DocumentFragment>
`;
